lib/setupmanagers/qemuhck/qemu_machine.rb

```diff
@@ -353,6 +352,11 @@ def device_machine_options
       @device_infos.map(&:machine_options).flatten.compact
     end
 
+    sig { returns(T::Array[String]) }
+    def device_iommu_device_param
+      @device_infos.map(&:iommu_device_param).flatten.compact
```

Comment:
Use `flat_map`.

Reply:
No, `iommu_device_param` is not an Array but just a mere String so we can just omit `flatten`.